app/controllers/batches_controller.rb

```diff
@@ -49,10 +51,14 @@ def create
       recipe.reload
       recipe.schedule.name.gsub!('Copy', "##{recipe.id}")
       recipe.name.gsub!('Copy', "##{recipe.id}")
+
       @batch.recipe = recipe
+        
+      if @batch.save(validate: false)
+        Recipe.update(recipe.id, :batch_id => @batch.id)
+        #why was this here? it undid all the recipe manipulation so batch.recipe was nil again.
+        #@batch.reload
```

Comment:
If you don't reload the object, the stuff on line 62 (formerly 57) would be nil. The point is that you save first, then refresh the object in memory from what was persisted.